### PR TITLE
mdns: setup CMake targets + modernize

### PR DESCRIPTION
**Pull request by prince-chrismc**
noticed https://github.com/mjansson/mdns/blob/e092f068b9cedccec06d040f3a0a950674060a7f/CMakeLists.txt#L33 was added

Specify library name and version:  **mdns/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

I noticed this needed TLC while searching for something (and I forgot what I was doing). I zapped the older unofficial releases

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mdns/all/conanfile.py [hunk 1]
```diff
@@ -1,35 +1,41 @@
...
         if str(self.settings.os) in ["Linux", "Android"]:
             self.cpp_info.system_libs.append('pthread')
 
-    def package_id(self):
-        self.info.header_only()
+        self.cpp_info.set_property("cmake_target_name", "mdns::mdns")
+
+        # TODO: to remove in conan v2 once cmake_find_package generators are removed
+        self.cpp_info.names["cmake_find_package"] = "mdns"
+        self.cpp_info.names["cmake_find_package_multi"] = "mdns"
```
Comment (SpaceIm) [inline recipes/mdns/all/conanfile.py:41]: useless

Comment (SpaceIm) [inline recipes/mdns/all/conanfile.py:41]:
```suggestion
        self.cpp_info.set_property("cmake_target_name", "mdns::mdns")
```

useless

File: recipes/mdns/all/conanfile.py [hunk 2]
```diff
@@ -1,35 +1,41 @@
...
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
 
     def package(self):
         self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
         self.copy(pattern="*.h", dst="include", src=self._source_subfolder)
 
+    def package_id(self):
+        self.info.header_only()
+
```
Comment (SpaceIm) [inline recipes/mdns/all/conanfile.py:30]: before source()

File: recipes/mdns/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/mdns/all/conanfile.py]: why add it to remove... fair enough

**Review by uilianries — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**